README.md

```diff
@@ -4,97 +4,55 @@ vim-slime-ext-plugins
 A "fork" of [vim-slime](https://github.com/jpalardy/vim-slime) to see how
 easy/hard it would be to keep target/language plugins outside of this codebase.
 
-## Proposed plugin structure
+This fork uses plugins to send data to its targets.
 
-Vim-slime needs a way to configure, and a way to send text. Configuration is
-done through a vim function, and sending through either a `system` call, or a
-vim function.
+## Plugin structure
 
-### Example of using a simple `system` call.
+Vim-slime needs a way to configure, and a way to send text. This is
+done through vim functions that you or your plugin must declare.
 
-In your configuration set 
+### Example of using a simple plugin
 
-```vimscript
-let slime_target_send="cat >> foo.txt"
-```
-
-You've effectively setup a vim-slime plugin that writes everything it's sent to
-'foo.txt' ! vim-slime will send any text it's supposed to send to `cat`'s STDIN.
+In this example, we are going to write a plugin that sends the text it receives to a file `foo.txt`. We need to define a function for plugin configuration, and another for handling the data.
 
-### Example of using a simple `system` call, but with some configuration.
+### Configuration
 
-Let's keep going with the previous plugin, but this time we want a bit of
-configuration. In `vim-slime` configuration is made on a per-buffer basis. We
-need to do that using a vim function. In you configuration file, you can add the
-following :
+The configuration function takes the current configuration in parameter and must return the updated configuration. For our example, one could write the following :
 
 ```vimscript
-function! SlimeSnitchConfig()
-  if !exists("b:slime_config['slime_snitch']")
-    let b:slime_config["slime_snitch"] = "snitch.txt"
+function! SlimeFooPluginConfig(config)
+  if !exists("a:config['foo']")
+    let a:config["foo"] = {"file": "foo.txt"}
   end
-  let b:slime_config["slime_snitch"] = input("Snitch filename : ", b:slime_config["slime_snitch"])
+  let a:config["foo"]["file"] = input("Target file: ", a:config["foo"]["file"])
+  return a:config
 endfunction
 ```
 
-then replace the target command with (the extra `bash -c` statement is needed to allow access to the $CONFIG
-environment variable set by vim-slime. See
-[here](https://unix.stackexchange.com/questions/126938/why-is-setting-a-variable-before-a-command-legal-in-bash))
-:
-```vimscript
-let slime_target_send = "bash -c 'tee $(echo $CONFIG | jq -r .slime_snitch)'"
-```
+In case your plugin does not need configuration, you can use the `slime#noop` convenience at the registration step.
```

Comment:
convenience -> convenience function ?

Reply:
whoopsie, this is fixed in [3ef523c](https://github.com/jpalardy/vim-slime-ext-plugins/pull/3/commits/3ef523cd9c58e961dc923bc89b3f00e33d850741) !